### DISCUSSION TITLE
kinematics

### DISCUSSION THREAD

**Discussion by GvBon** [phil-barrett/grblHAL-teensy-4.x#77] [General]
Love grblHAL, got it working in no time with IOsender. Thanks for all the great work!

For two of my machines I need to change the kinematics. I have been using EdingCNC for 12 years now with my three machines but would love to transfer to grblHAL. But two are not cartesian. One of the axes is polar, the other cartesian (check http://www.gijsvanbon.nl/scriptum.html ).
As the source code is not public but must be build through web builder there are only three options. How could I use my own kinematic functions?

**Comment by phil-barrett** [comment]
The code is, in fact, public.  It is a little complex to set up as you need to assemble the various libraries.  There are separate repositories for the core, drivers and plug ins.  [Start here.](https://github.com/grblHAL/core).  Take a look at the coreXY file in the core code as that is an example of a different kinematic system and should give you a bread-crumb trail to follow.

[edit] the discussion section of the core repo is a great place to ask questions.  Terje is usually very responsive and helpful.

**Comment by GvBon** [comment]
Thanks Phil for your reply!
Good to know it is public. I will (try to) dive in...
